draft-ietf-taps-arch.md

```diff
@@ -325,90 +335,90 @@ The diagram below provides a high-level view of the actions taken during the lif
 
-* Send: The action to transmit a Message or partial Message over a Connection to a Remote Endpoint. The interface to Send may include options specific to how the Message's content is to be sent. Status of the Send operation may be delivered back to the application in an event ({{events}}).
+* Send: The action to transmit a Message or partial Message over a Connection to the remote system. The interface to Send MAY include options specific to how the Message's content is to be sent. Status of the Send operation can be delivered back to the application in an event ({{events}}).
 
-* Receive: An action that indicates that the application is ready to asynchronously accept a Message over a Connection from a Remote Endpoint, while the Message content itself will be delivered in an event ({{events}}). The interface to Receive may include options specific to the Message that is to be delivered to the application.
+* Receive: An action that indicates that the application is ready to asynchronously accept a Message over a Connection from a remote system, while the Message content itself will be delivered in an event ({{events}}). The interface to Receive MAY include options specific to the Message that is to be delivered to the application.
 
 ### Event Handling {#events}
 
-This list of events that can be delivered to an application is not exhaustive, but gives the top-level categories of events. The API may expand this list.
+This list of events that can be delivered to an application is not exhaustive, but gives the top-level categories of events. The API MAY expand this list.
 
 * Connection Ready: Signals to an application that a given Connection is ready to send and/or receive Messages. If the Connection relies on handshakes to establish state between peers, then it is assumed that these steps have been taken.
 
-* Connection Finished: Signals to an application that a given Connection is no longer usable for sending or receiving Messages. This should deliver an error to the application that describes the nature of the termination.
+* Connection Finished: Signals to an application that a given Connection is no longer usable for sending or receiving Messages. The event SHOULD deliver an error to the application that describes the nature of the termination.
```

Comment:
Why an "error"? If it can also be due to a graceful shutdown, then "notification" may be a better word.

Reply:
Made this "reason or error"